src/convertToObject.js

```diff
@@ -1,12 +1,18 @@
 'use strict';
 
-/**
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = sourceString
+    .split(';')
+    .map((element) => element.split(':'))
+    .map((element) => element.map((str) => str.trim()))
+    .filter((element) => element.length > 1)
+    .reduce((res, [key, value]) => {
```

Comment:
[CODE KNOWLEDGE] - Don't use iteration methods to modify an already existing array/object. `reduce` is an exception to this rule, as it is used correctly here to transform an array into an object.